plutus-ir/test/TransformSpec.hs

```diff
@@ -16,136 +15,8 @@ transform = testNested "transform" [
     ]
 
 thunkRecursions :: TestNested
-thunkRecursions = testNested "thunkRecursions" [
-    goldenPir "listFold" listFold,
-    goldenPir "monoMap" monoMap
+thunkRecursions = testNested "thunkRecursions"
+    $ map (goldenPir (runQuote . thunkRecursionsTerm) term)
+    [ "listFold"
+    , "monoMap"
     ]
-
-listFold :: Term TyName Name ()
-listFold = runQuote $ thunkRecursionsTerm =<< do
-    {-
-    This implements foldl:
-
-        foldl : (a -> b -> a) -> a -> [b] -> a
-        foldl f acc lst = case lst of
-            [] -> acc
-            x:xs -> foldl f (f acc x) xs
-    -}
-    lb@(Datatype _ d _ destr _) <- listDatatype
-    avd <- do
-        a <- freshTyName () "a"
-        pure $ TyVarDecl () a (Type ())
-    bvd <- do
-        b <- freshTyName () "b"
-        pure $ TyVarDecl () b (Type ())
-
-    let fty = mkIterTyFun () [ mkTyVar () avd, mkTyVar () bvd ] (mkTyVar () avd)
-    let accTy = mkTyVar () avd
-    let listBTy = TyApp () (mkTyVar () d) (mkTyVar () bvd)
-
-    let foldTy =
-            mkIterTyForall () [avd, bvd] $
-            mkIterTyFun () [fty, accTy, listBTy]
-            (mkTyVar () avd)
-
-    fvd <- do
-        fn <- freshName () "foldl"
-        pure $ VarDecl () fn foldTy
-    fld <- do
-        f <- freshName () "f"
-        acc <- freshName () "acc"
-        lst <- freshName () "lst"
-        x <- freshName () "x"
-        xs <- freshName () "xs"
-        pure $
-            LamAbs () f fty $
-            LamAbs () acc accTy $
-            LamAbs () lst listBTy $
-            -- match {a}
-            mkIterApp () (TyInst () (Var () destr) (mkTyVar () avd))
-            [
-                -- lst
-                Var () lst,
-                -- Nil case
-                -- acc
-                Var () acc,
-                -- Cons case
-                -- \(x:b) (xs::[b]) -> foldl f (f acc x) xs
-                LamAbs () x (mkTyVar () bvd) $
-                LamAbs () xs listBTy $
-                mkIterApp () (mkVar () fvd)
-                [
-                    Var () f,
-                    mkIterApp () (Var () f)
-                    [
-                        Var () acc,
-                        Var () x
-                    ],
-                    Var () xs
-                ]
-            ]
-
-    pure $
-        Let () Rec [ DatatypeBind () lb ] $
-        Let () Rec [ TermBind () fvd fld ] $ mkVar () fvd
-
-monoMap :: Term TyName Name ()
-monoMap = runQuote $ thunkRecursionsTerm =<< do
-    {-
-    This implements a monomorphic map (which does not need to be thunked):
-
-        map : (1 -> 1) -> List 1 -> List 1
-        map f lst = case lst of
-            [] -> []
-            x:xs -> f x : map f xs
-    -}
-    lb@(Datatype _ d _ destr [nil, cons]) <- listDatatype
-
-    let elemTy = TyInt () 1
-
-    let fty = TyFun () elemTy elemTy
```

Comment:
Ouch. What do we do here? `TyFun` expects two types, but is applied to two sizes instead (yes, `TyInt` is a terribly confusing way to say "size constant"). Looking at the comment I see 

`map : (1 -> 1) -> List 1 -> List 1`

which suggests that it's intentional, but why? @michaelpj

Reply:
I was feeling lazy and I wanted something that would be a monomorphic map without having to define a bunch more stuff. I'd just replace that with a more sensible type.